code/frontend/vite.config.ts

```diff
@@ -1,8 +1,11 @@
 import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 
+const envDir = process.env.VITE_ENV_DIR;
```

Comment:
Apologies if I misunderstand the code here. Are we now tying the typescript app in to having to have an .env file read in rather than having the hosting process deal with that? At the moment whether the code runs in App Service or in a Container or locally; the host/process has environment variables setup before the app runs - does this mean you know have to copy an .env in at build time?

Reply:
We are injecting a configuration into the typescript app at build time so that it is embedded into the frontend app when it is required. 

Now, we could also just have the frontend API layer read the env and return it via an API endpoint - that is another approach. I did think about that approached and picked this one since it would mean one less API call. I'm happy to reconsider if you think my current approach is making build-time more complicated. Thoughts?